storage/file.go

```diff
@@ -86,6 +89,18 @@ func (f FileServiceClient) DeleteShareIfExists(name string) (bool, error) {
 // deleteShare makes the call to Delete Share operation endpoint and returns
 // the response
 func (f FileServiceClient) deleteShare(name string) (*storageResponse, error) {
+	if err := f.checkForStorageEmulator(); err != nil {
```

Comment:
Yeah, I know it is not the most elegant thing :(
Still, if a user doesn't know files aren't supported, the error while doing the file operations would be pretty confusing.